services/worker/src/worker/job_runners/dataset_split_names.py

```diff
@@ -0,0 +1,191 @@
+
+
+class DatasetSplitNamesJobRunnerError(JobRunnerError):
+    """Base class for split names job runner exceptions."""
+
+    def __init__(
+        self,
+        message: str,
+        status_code: HTTPStatus,
+        code: DatasetSplitNamesJobRunnerErrorCode,
+        cause: Optional[BaseException] = None,
+        disclose_cause: bool = False,
+    ):
+        super().__init__(
+            message=message, status_code=status_code, code=code, cause=cause, disclose_cause=disclose_cause
+        )
+
+
+class PreviousStepStatusError(DatasetSplitNamesJobRunnerError):
+    """Raised when the previous step gave an error. The job should not have been created."""
+
+    def __init__(self, message: str, cause: Optional[BaseException] = None):
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "PreviousStepStatusError", cause, False)
+
+
+class PreviousStepFormatError(DatasetSplitNamesJobRunnerError):
+    """Raised when the content of the previous step has not the expected format."""
+
+    def __init__(self, message: str, cause: Optional[BaseException] = None):
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "PreviousStepFormatError", cause, False)
+
+
+class ResponseNotReadyError(DatasetSplitNamesJobRunnerError):
+    """Raised when the response has not been processed yet from any source."""
+
+    def __init__(self, message: str):
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "ResponseNotReady")
+
+
+class PendingJob(TypedDict):
+    dataset: str
+    config: Optional[str]
+
+
+class SuccessJob(PendingJob):
+    split: str
+
+
+class FailedJob(PendingJob):
+    error: Mapping[str, Any]
+
+
+class DatasetSplitNamesResponseContent(TypedDict):
+    splits: List[SuccessJob]
+    pending: List[PendingJob]
+    failed: List[FailedJob]
+
+
+def get_response_from_sources(
+    config_name: str, streaming_source: List[CacheEntryWithInfo], dataset_info_source: List[CacheEntryWithInfo]
+) -> CacheEntryWithInfo:
+    result_from_streaming = next((item for item in streaming_source if item["config"] == config_name), None)
+    last_result = None
+    if result_from_streaming:
+        if result_from_streaming["http_status"] == HTTPStatus.OK:
+            return result_from_streaming
+        last_result = result_from_streaming
+    result_from_dataset_info = next((item for item in dataset_info_source if item["config"] == config_name), None)
+    if result_from_dataset_info:
+        if result_from_dataset_info["http_status"] == HTTPStatus.OK:
+            return result_from_dataset_info
+        last_result = result_from_dataset_info
+    if last_result:
+        return last_result
+
+    raise ResponseNotReadyError("Not ready yet")
+
+
+def compute_dataset_split_names_response(
+    dataset: str,
+) -> DatasetSplitNamesResponseContent:
+    """
+    Get the response of /dataset-split-names for one specific dataset on huggingface.co
```

Comment:
I will change to /splits, I think it is better to have in the comment where is this used because it is redundant as we have already the cache_kind/job_type in get_job_type.